tests/enum_input.rs

```diff
@@ -0,0 +1,133 @@
+use rand::{prelude::ThreadRng, thread_rng, Rng};
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use std::net::SocketAddr;
+
+use ggrs::{Config, Frame, GGRSRequest, GameStateCell, InputStatus};
+
+fn calculate_hash<T: Hash>(t: &T) -> u64 {
+    let mut s = DefaultHasher::new();
+    t.hash(&mut s);
+    s.finish()
+}
+
+pub struct GameStub {
+    pub gs: StateStub,
+}
+use bytemuck::{CheckedBitPattern, NoUninit, Zeroable};
+
+#[repr(u8)]
+#[derive(Copy, Clone, PartialEq, CheckedBitPattern, NoUninit)]
+pub enum EnumInput {
+    Val1,
+    Val2,
+}
```

Comment:
For now, it only supports fieldless enums, because NoUninit doesn't support fields, it might be technically possible to add this support upstream in bytemuck ?

Reply:
I'm wondering if NoUninit is actually needed, and if yes, could MaybeUninit replace it ? 🤔 I'm gonna explore this idea.

Reply:
Please do! Sounds exciting! :)

Reply:
I gave it a small attempt, I think it's a good direction, but our problem is [bytes_of](https://docs.rs/bytemuck/latest/bytemuck/fn.bytes_of.html) used here https://github.com/gschup/ggrs/blob/0d70f4524b02a945999cdfed83a111e8c1140ed0/src/network/protocol.rs#L54, which has a type restriction on `NoUninit`. I *feel* like it should the contraint could be relaxed to support enum with fields, but it might need quite some work from bytemuck, such as https://github.com/Lokathor/bytemuck/issues/55.

I'm gonna create an issue on bytemuck and see where it goes from there.

Reply:
Since I was reading thru this PR anyway, just a small note: `bytes_of` is used elsewhere so this doesn't solve the problem, but instead of creating a `zeroed` object just to call `bytes_of(&zeroed).len() * num_players` here, you could forego all of that and simply call `core::mem::size_of::<T::Input>() * num_players` to get the size

Reply:
thanks for the comment! good catch, that definitely seems like a smarter solution.